benchmark_runner/common/oc/oc.py

```diff
@@ -441,6 +440,33 @@ def get_worker_nodes(self):
         """
         return self.run(fr""" {self.__cli} get nodes -l node-role.kubernetes.io/worker= -o jsonpath="{{range .items[*]}}{{.metadata.name}}{{'\n'}}{{end}}" """)
 
+    @staticmethod
+    @typechecked
+    def check_node_status(nodes_list: list):
+        """
+        This method check node status
+        @param nodes_list:
+        @return: True when all nodes in ready status
+        """
+        # Check if any node is not in 'Ready' status
+        for node in nodes_list:
+            node_name, node_status = node.split()
+            if node_status != 'Ready':
+                raise NodeNotReady(node_name, node_status)
```

Comment:
Cleaner to return False here, and let the caller decide whether to raise an exception.

Reply:
I think we must raise error if one of the node is not ready and CI failure